build/docker/kmesh-compiler.dockerfile

```diff
@@ -0,0 +1,36 @@
+#
+# Dockerfile for building openEuler kmesh docker image.
+# 
+# Usage:
+# docker build -f kmesh.dockerfile -t kmesh:latest .
+# docker run -itd --privileged=true -v /usr/src:/usr/src -v /usr/include/linux/bpf.h:/kmesh/config/linux-bpf.h -v /etc/cni/net.d:/etc/cni/net.d -v /opt/cni/bin:/opt/cni/bin -v /mnt:/mnt -v /sys/fs/bpf:/sys/fs/bpf -v /lib/modules:/lib/modules --name kmesh kmesh:latest
+#
+
+# base image
+FROM openeuler/openeuler:23.09
+
+# Setup Go
+COPY --from=golang:1.21.6 /usr/local/go/ /usr/local/go/
```

Comment:
we should not depend on host golang

Reply:
run install during the build

Reply:
> we should not depend on host golang

This will be changed later and does not affect the current pr

Reply:
> run install during the build
This will be changed later and does not affect the current pr